src/Atoms/UserProfileData/UserProfileData.js

```diff
@@ -0,0 +1,82 @@
+import propTypes from 'prop-types';
+import { useContext, useEffect, useState } from 'react';
+
+import UserProfileDataStyled from './UserProfileDataStyled';
+import participants from '../../Data/MongoDBMock/participants';
+import { UserDataContext } from '../../Context/UserDataContext';
+
+// import { useLocation } from 'react-router-dom';
+
+const initialData = {
+  address: {
+    country: 'Polska',
+    city: 'Sfornegacie',
+    street: 'ul.Pszczelna',
+    numberOfHouse: '27/5',
+    postalCode: '50-124',
+  },
+};
+
+const UserProfileData = () => {
+  const { state } = useContext(UserDataContext);
+  const { userId, userName, userSurname, isAuthenticated } = state;
+  const [userObject, setUserObject] = useState(initialData);
+
+  if (!isAuthenticated) {
+    throw new Error('Your not allowed to be here!');
```

Comment:
Maybe it could redirect to a different page instead of throwing error?

Reply:
Login page Maybe ?